### PR TITLE
Reset captive status on respawn for reliability

### LINKED ISSUE #1849: Captives persist after death
ACE3 Version: 3.1.1

Happens on Stable, Dev and RC Builds of ArmA

**Mods:**
* @cba_a3
* @ace3

If a player who is restrained by zip ties dies or respawns they remain restrained and can not be unrestrained. Requires them to re-log onto server to fix the problem

**Steps to reproduce:**
1) Zip tie another player
2) Kill the zip-tied player or have them respawn
When the zip-tied player respawns the are permanently zip tied until they log off the server

Multiplayer Bug

*edit* when my friends get on tomorrow I can record the problem if that helps*

### PR DESCRIPTION
**Pull request by SilentSpike**
Previously we were resetting on death, but this caused problems when the respawnDelay was 0 as the dead unit wasn't actually the player.

To account for "group" and "side" respawn types where you change unit on respawn and could potentially become a captive unit, I am explicity checking the mission respawn type before resetting variables.

I believe this should fix #1849 

Could use more testing as I wasn't entirely sure what to look for

### CODE REVIEW

File: addons/captives/functions/fnc_handleRespawn.sqf [hunk 1]
```diff
@@ -16,24 +16,40 @@
  */
 #include "script_component.hpp"
 
+systemChat str(_this);
```
Comment (commy2) [inline addons/captives/functions/fnc_handleRespawn.sqf]: reminder

**Comment by Glowbal** [comment]
Status?

**Comment by PabstMirror** [comment]
:+1:

**Comment by commy2** [comment]
Fixes the respawn bug on the normal respawn. No idea how the group respawn is supposed to work.
:+1:

**Pull request merged**